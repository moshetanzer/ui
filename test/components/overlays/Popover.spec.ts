import { describe, it, expect } from 'vitest'
import { UPopover } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('Popover', () => {
  it.each([
    ['basic case', { props: { modelValue: true } }]
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof UPopover.props>) => {
    const html = await ComponentRender(nameOrHtml, options, UPopover)
    expect(html).toMatchSnapshot()
  })
})
