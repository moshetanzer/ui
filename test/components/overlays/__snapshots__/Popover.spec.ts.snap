// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Popover > renders basic case correctly 1`] = `
"<!-- eslint-disable-next-line vue/no-template-shadow -->
<div class="relative" modelvalue="true" data-headlessui-state="">
  <div class="inline-flex w-full" role="button" id="headlessui-popover-button-0" aria-expanded="false" data-headlessui-state=""><button> Open </button></div>
  <!---->
  <!--v-if-->
  <!--v-if-->
</div>
<div style="position: fixed; height: 0px; padding: 0px; overflow: hidden; clip: rect(0, 0, 0, 0); white-space: nowrap; border-width: 0px; display: none;"></div>"
`;
